presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -227,4 +233,38 @@ public static ParquetPageSource createParquetPageSource(
             throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, message, e);
         }
     }
+
+    public static TupleDomain<ColumnDescriptor> getParquetTupleDomain(Map<List<String>, RichColumnDescriptor> descriptorsByPath, TupleDomain<HiveColumnHandle> effectivePredicate)
```

Comment:
Any reason why you moved these methods from the `PredicateUtils` class to here?

Reply:
it is mostly just called here. Try to not have cross module dependency

Reply:
This module already depends on `presto-parquet`, also this class already imports `PredicateUtils` methods. I think we can keep these methods in their original class, the `PredicateUtils`.

Reply:
oh, just found, HiveColumnHandle could not include in presto-parquet module, so I leave it here.